### PR TITLE
Added requestedID for each call with fresh open session

### PR DESCRIPTION
**Pull request by arahamad**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -92,16 +101,26 @@ func main() {
...
 			fmt.Println("You selected choice to get volume details")
 			fmt.Printf("Please enter volume ID: ")
 			_, er11 = fmt.Scanf("%s", &volumeID)
 			volume, errr := sess.GetVolume(volumeID)
 			if errr == nil {
-				logger.Info("SUCCESSFULLY get volume details ================>", zap.Reflect("VolumeDetails", volume))
+				ctxLogger.Info("SUCCESSFULLY get volume details ================>", zap.Reflect("VolumeDetails", volume))
 			} else {
-				logger.Info("FAILED to get volume details ================>", zap.Reflect("VolumeID", volumeID), zap.Reflect("Error", errr))
+				errr = updateRequestID(errr, requestID)
```
Comment (prgavali) [inline main.go]: `errr = updateRequestID(errr, requestID)`  - why we are updating requestId explicitly? (edited) 
I guess , `volume, errr := sess.GetVolume(volumeID)` should return error with `request id` 
All the library API should return error with resuestID. So driver code  do not have to worry about that . The driver will just pass in requestID to API  through sesssion
Also , the error object belongs to provider package(volume-provider/vpc) , So let all the properties of error be set from provider only .

File: main.go [hunk 2]
```diff
@@ -42,6 +43,24 @@ func main() {
...
 	)
 	logger := zap.New(core, zap.AddCaller())
+	return logger, traceLevel
+}
+
+func updateRequestID(err error, requestID string) error {
+	if err == nil {
+		return err
+	}
+	usrError, ok := err.(userError.Message)
```
Comment (prgavali) [inline main.go]: The datatype  check are  very good from  programming perspective . should be used minimum if we do not have choice. If we set request id in the provider code , this code is not required

Comment (prgavali) [reply main.go]: Considering the  no. of places the change need to be made , we are fine with the current code

File: provider/utils/init_provider.go [hunk 3]
```diff
@@ -86,26 +86,28 @@ func isEmptyStringValue(value *string) bool {
 }
 
 // OpenProviderSession ...
-func OpenProviderSession(conf *config.Config, providers registry.Providers, providerID string, logger *zap.Logger) (session provider.Session, fatal bool, err1 error) {
+func OpenProviderSession(conf *config.Config, providers registry.Providers, providerID string, logger *zap.Logger) (session provider.Session, fatal bool, errReturn error) {
 	prov, err := providers.Get(providerID)
```
Comment (narkarum) [inline provider/utils/init_provider.go]: we do not need errReturn and err separately.
We can assign to err itself,

Comment (arahamad) [reply provider/utils/init_provider.go]: errReturn is for returning from the method, and we can't use  errReturn instead of err

Comment (narkarum) [reply provider/utils/init_provider.go]: https://github.com/IBM/ibmcloud-storage-volume-lib/blob/master/volume-providers/vpc/provider/create_volume.go#L27-L30

Comment (narkarum) [reply provider/utils/init_provider.go]: It is not required.

Comment (arahamad) [reply provider/utils/init_provider.go]: > It is not required.

Done

**Pull request merged**